sotabencheval/language_modelling/wikitext.py

```diff
@@ -0,0 +1,132 @@
+import os
+from itertools import islice
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+
+from sotabenchapi.check import in_check_mode
+from sotabenchapi.client import Client
+from sotabenchapi.core import BenchmarkResult, check_inputs
+from sotabencheval.core import BaseEvaluator
+from sotabencheval.utils import calculate_batch_hash, extract_archive, change_root_if_server, is_server
+
+def to_numpy(*args):
+    def convert(a):
+        if hasattr(a, 'cpu') and hasattr(a, 'numpy'):
+            return a.cpu().numpy()
+        if isinstance(a, list):
+            return np.array(a)
+        return a 
+    return [convert(a) for a in args]
+
+class WikiTextDataset(Enum):
+    WikiText103 = ('WikiText-103', 245569, 267735)
+    WikiText2 = ('WikiText-2', 245569, 33278)
+    
+    def __init__(self, pwc_name, testset_size, vocab_size):
+        self.pwc_name = pwc_name
+        self.testset_size = testset_size
+        self.vocab_size = vocab_size
+    
+    def get_path(self, local_root, local_unzip=False):
+        root = Path(change_root_if_server(root=local_root,
+                                          server_root=".data/nlp/" + self.pwc_name.lower()))
+        zip_name = self.pwc_name.lower() + "-v1.zip"
+        dataset_path = root / "wiki.test.tokens"
+        if not dataset_path.exists(): # unzip
+            extract_archive(str(root / zip_name), to_path=root.parent)
+        return dataset_path
+
+
+def gether_probs(log_probabilities, targets):
```

Comment:
?

Reply:
Depending on the level of elasticity in the model code one can use 3 levels of validation(the 
 validation checks are yet to be written though).
- level 1: you just provide a summed log probability over a given batch of sentences (batch of sentences are in targets)
- level 2: if your model lets you get a probablity for each word /token in the batch that is even better we accept that.
- level 3:  but ideally we would like to see your prediction of log probabilities of all possible words for each word in a batch. os we allow the input to be of shape: `[batch_size x seq_length x vocab_size]`. There we need the gather method implemented in `np` and `torch` (For speed).

Reply:
Okay, just needs to be documented (docstrings, and example usage in the docs themselves).